replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/ReplicationItemManagerImpl.java

```diff
@@ -54,15 +58,22 @@
 
   private final PersistentStore persistentStore;
 
+  // helps with solr commit time slowness.
+  private Map<String, ReplicationItem> cache =
+      ExpiringMap.builder().expiration(45, TimeUnit.SECONDS).build();
+
   public ReplicationItemManagerImpl(PersistentStore persistentStore) {
     this.persistentStore = persistentStore;
   }
 
   @Override
   public Optional<ReplicationItem> getItem(String metadataId, String source, String destination) {
+    if (cache.containsKey(metadataId + source + destination)) {
+      return Optional.of(cache.get(metadataId + source + destination));
+    }
     String cqlFilter =
         String.format(
-            "'id' = '%s' AND 'source' = '%s' AND 'destination' = '%s'",
+            "'mcard-id' = '%s' AND 'source' = '%s' AND 'destination' = '%s'",
```

Comment:
What is the distinction here between `mcard-id` and `metadataId`?

Reply:
They are the same thing here.  Poor naming choice on my part. I will fix it.